app/components/Blockchain/Transaction/index.js

```diff
@@ -17,7 +18,9 @@ const mapDispatchToProps = dispatch => ({
     dispatch(showModal(MODAL_TYPES.ADD_CONTACT, props))
 })
 
-const mapContactsDataToProps = (contacts: Object) => ({ contacts })
+const mapContactsDataToProps = (contacts: Object) => ({
+  contacts: invert(contacts)
+})
```

Comment:
Since `contacts` is a `{ name: address}` dict, instead of looping it to find key by value, I invert it to be `{ address: name }` and now all you need is `name = contacts[address]`🏎

Reply:
Haha this is really clever never used lodash invert before 🔥